docs/core/v3.0.rst

```diff
@@ -932,25 +936,21 @@ containing the following names:
     version number string to help with discovery of this
     specification.
 
-``metadata_encoding``
-^^^^^^^^^^^^^^^^^^^^^
-
-    A string containing the URI pointing to a document describing the method
-    used for encoding group and array metadata documents.
-
-    For document using the default JSON encoding and format describe in this document
-    then the value must be ``"https://purl.org/zarr/spec/core/3.0``.
-
 ``metadata_key_suffix``
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-    A string containing a suffix to add to the metadata keys when saving into
-    the store. By default ``".json"``.
+    A string containing a suffix to add to the array and group metadata keys
```

Comment:
What do you see as the advantage of using `metadata_key_suffix` rather than `metadata_encoding` to identify the metadata encoding?  I think it may be clearer to specify `metadata_encoding` and instead eliminate `metadata_key_suffix`, since in principle the same extension could be used for two different encodings, etc. 

However, I actually think we should just eliminate metadata_encoding for now, and can worry about it later if there is a concrete proposal for an alternative metadata encoding that someone wants to add.

Reply:
I think we should just remove both for now.  We can still later add a `metadata_encoding` attribute if we want, rather than adding it under "extensions" --- since any volume using a different `metadata_encoding` will only be usable by implementations that support it, the default behavior of failing when encountering any unknown metadata members will do the right thing.

As far as providing a hint to implementations that the extension may change --- I can't imagine it being very difficult to change that later in any implementation.  In general when writing software I think trying to add additional abstraction layers/indirection in anticipation of *future* extensions that you haven't already fully thought through isn't that helpful --- you are likely as not to guess wrong, and just end up doing extra work and having extra complexity from unused additional indirection/abstraction layers.

Reply:
Also referencing #37, where exactly such an extension point was agreed on. I'd rather keep this in atm and tend to making `metadata_encoding` a regular extension point.

> As far as providing a hint to implementations that the extension may change --- I can't imagine it being very difficult to change that later in any implementation. In general when writing software I think trying to add additional abstraction layers/indirection in anticipation of future extensions that you haven't already fully thought through isn't that helpful --- you are likely as not to guess wrong, and just end up doing extra work and having extra complexity from unused additional indirection/abstraction layers.

Agreed, but since this topic came up multiple times (#37, #141), and the extension interface is still quite generic, I think it's worth a try. The only difference is to make this an explicit extension point for metadata encoding, instead of an arbitrary array extension.

Reply:
It probably was a mistake to mix this topic with other rather smaller things in this PR. I'll go forward with `metadata_encoding` as an extension for now, but opened up a separate issue to discuss if this should be removed #174